File: Data-Structures-Algorithms/Strings/valid-palindrome.java
// https://leetcode.com/problems/valid-palindrome/submissions/?envType=study-plan-v2&envId=top-interview-150

class Solution {
    public boolean isPalindrome(String s) {
        s = s.replaceAll("[^A-Za-z0-9]", "").toUpperCase();
        for (int i = 0; i < (s.length() )/ 2; i++) {
            if (s.charAt(i) != s.charAt(s.length() - i - 1)) {
                return false;
            }
        }
        return true;
    }
}

File: Data-Structures-Algorithms/Strings/valid_palindrome.py
# https://leetcode.com/problems/valid-palindrome/

# https://neetcode.io/problems/string-encode-and-decode?list=neetcode150

class Solution:
    def isPalindrome(self, strs: str) -> bool:
        s = 0
        n = len(strs)
        e = n - 1
        
        while s < e:
            if not strs[s].isalnum():
                s += 1
                continue
            if not strs[e].isalnum():
                e -= 1
                continue
            if strs[s].lower() != strs[e].lower():
                return False
            else:
                s += 1
                e -= 1
    
        return True
        
